.github/workflows/nightly-build-publish.yml

```diff
@@ -18,29 +18,55 @@ jobs:
   nightly-build-publish:
     runs-on: ubuntu-latest
     steps:
-      - name: Clone source code
+      - 
+        name: Clone source code
         uses: actions/checkout@v1
         with:
           fetch-depth: 1
-
-      - name: Login to quay.io
+      - 
+        name: Prepare
+        id: prep
+        run: |
+          SHORT_SHA1=$(git rev-parse --short HEAD)
+          echo ::set-output name=short_sha1::${SHORT_SHA1}
+          VERSION=$(head -n 1 VERSION)
+          echo ::set-output name=version::${VERSION}
+          IMAGE=che-devfile-registry
+          echo ::set-output name=image::${IMAGE}
+          PLATFORMS=$(cat PLATFORMS)
+          echo ::set-output name=platforms::${PLATFORMS}
+      -
+        name: "Set up QEMU"
+        uses: docker/setup-qemu-action@v1
+      -
+        name: "Set up Docker Buildx"
+        uses: docker/setup-buildx-action@v1
+      -
+        name: "Docker quay.io Login"
         uses: docker/login-action@v1
         with:
           registry: quay.io
           username: ${{ secrets.QUAY_USERNAME }}
           password: ${{ secrets.QUAY_PASSWORD }}
+      - 
+        name: Login to docker.io
+        uses: docker/login-action@v1
+        with:
+          registry: docker.io
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
 
       - name: Build and push base images
         run: /bin/bash arbitrary-users-patch/build_images.sh --push
 
       - name: Build and push happy path image
         run: /bin/bash arbitrary-users-patch/happy-path/build_happy_path_image.sh --push
-
-      - name: Build and push devfile image
-        run: |
-          SHORT_SHA1=$(git rev-parse --short HEAD)
-          docker build -t che-devfile-registry -f ./build/dockerfiles/Dockerfile --target registry .
-          docker tag che-devfile-registry quay.io/eclipse/che-devfile-registry:nightly
-          docker push quay.io/eclipse/che-devfile-registry:nightly
-          docker tag che-devfile-registry quay.io/eclipse/che-devfile-registry:${SHORT_SHA1}
-          docker push quay.io/eclipse/che-devfile-registry:${SHORT_SHA1}
+      -
+        name: "Build and push"
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./build/dockerfiles/Dockerfile
+          platforms: ${{ steps.package.outputs.content }}
```

Comment:
@nickboldt Image is still single arch because I guess slight change is required here. Instead of `platforms: ${{ steps.package.outputs.content }}`, it needs to be `platforms: ${{ steps.prep.outputs.platforms }}` as defined on line 37.